src/main/resources/META-INF/rewrite/junit5.yml

```diff
@@ -57,6 +57,12 @@ preconditions:
   - org.openrewrite.java.search.DoesNotUseType:
       fullyQualifiedTypeName: org.testng..*
       includeImplicit: true
+  - org.openrewrite.gradle.search.DoesNotIncludeDependency:
+      groupId: org.testng
+      artifactId: testng
+  - org.openrewrite.maven.search.DoesNotIncludeDependency:
+      groupId: org.testng
+      artifactId: testng
```

Comment:
Could we introduce an overarching recipe?
`org.openrewrite.java.dependencies.DoesNotIncludeDependency`?